src/Formats/JSONExtractTree.cpp

```diff
@@ -0,0 +1,1574 @@
+                        return false;
+                    }
+                }
+            }
+            else
+            {
+                for (const auto & [key, value] : object)
+                {
+                    auto index = name_to_index_map.find(key);
+                    if (index != name_to_index_map.end())
+                    {
+                        if (nested[index->second]-> insertResultToColumn(tuple.getColumn(index->second), value, insert_settings, format_settings, error))
+                        {
+                            were_valid_elements = true;
+                        }
+                        else if (!insert_settings.insert_default_on_invalid_elements_in_complex_types)
+                        {
+                            set_size(old_size);
+                            error += fmt::format("(during reading tuple element \"{}\")", key);
+                            return false;
+                        }
+                    }
+                }
+            }
+
+            set_size(old_size + static_cast<size_t>(were_valid_elements));
+            return were_valid_elements;
+        }
+
+        error = fmt::format("cannot read Tuple value from JSON element: {}", jsonElementToString<JSONParser>(element, format_settings));
+        return false;
+    }
+
+private:
+    std::vector<std::unique_ptr<JSONExtractTreeNode<JSONParser>>> nested;
+    std::vector<String> explicit_names;
+    std::unordered_map<std::string_view, size_t> name_to_index_map;
+};
+
+template <typename JSONParser>
+class MapNode : public JSONExtractTreeNode<JSONParser>
+{
+public:
+    explicit MapNode(std::unique_ptr<JSONExtractTreeNode<JSONParser>> value_) : value(std::move(value_)) { }
+
+    bool insertResultToColumn(
+        IColumn & column, const typename JSONParser::Element & element, const JSONExtractInsertSettings & insert_settings, const FormatSettings & format_settings, String & error) const override
+    {
+        if (!element.isObject())
+        {
+            error = fmt::format("cannot read Map value from JSON element: {}", jsonElementToString<JSONParser>(element, format_settings));
+            return false;
+        }
+
+        auto & map_col = assert_cast<ColumnMap &>(column);
+        auto & offsets = map_col.getNestedColumn().getOffsets();
+        auto & tuple_col = map_col.getNestedData();
+        auto & key_col = tuple_col.getColumn(0);
+        auto & value_col = tuple_col.getColumn(1);
+        size_t old_size = tuple_col.size();
+
+        auto object = element.getObject();
+        auto it = object.begin();
+        for (; it != object.end(); ++it)
+        {
+            auto pair = *it;
+
+            /// Insert key
+            key_col.insertData(pair.first.data(), pair.first.size());
+
+            /// Insert value
+            if (!value-> insertResultToColumn(value_col, pair.second, insert_settings, format_settings, error))
```

Comment:
```suggestion
            if (!value->insertResultToColumn(value_col, pair.second, insert_settings, format_settings, error))
```

Maybe could you please format the whole file with `clang-format?`

Reply:
Sure

Reply:
`value-> insertResultToColumn`
Seems like I made a mistake during some find and replace :)